psse/psse-model/src/main/java/com/powsybl/psse/model/PsseRawReader.java

```diff
@@ -85,122 +321,108 @@ public void handleError(DataProcessingException error, Object[] inputRow, Parsin
         return records;
     }
 
-    private PsseCaseIdentification readCaseIdentification(BufferedReader reader) throws IOException {
-        String line = readLineAndRemoveComment(reader);
-        PsseCaseIdentification caseIdentification = parseRecord(line, PsseCaseIdentification.class);
-        caseIdentification.setTitle1(reader.readLine());
-        caseIdentification.setTitle2(reader.readLine());
-        return caseIdentification;
+    private static String removeComment(String line) {
+        int slashIndex = line.lastIndexOf('/');
+        if (slashIndex == -1) {
+            return line;
+        }
+        return line.substring(0, slashIndex);
     }
 
-    private List<PsseTransformer> readTransformers(BufferedReader reader) throws IOException {
-        List<PsseTransformer> transformers = new ArrayList<>();
-
-        List<String> records = readRecordBlock(reader);
-        int i = 0;
-        while (i < records.size()) {
-            PsseTransformer transformer = new PsseTransformer();
-            transformer.setFirstRecord(parseRecord(records.get(i++), PsseTransformer.FirstRecord.class));
-            transformer.setSecondRecord(parseRecord(records.get(i++), PsseTransformer.SecondRecord.class));
-            transformer.setThirdRecord1(parseRecord(records.get(i++), PsseTransformer.ThirdRecord.class));
-            transformer.setThirdRecord2(parseRecord(records.get(i++), PsseTransformer.ThirdRecord.class));
-            if (transformer.getFirstRecord().getK() != 0) {
-                transformer.setThirdRecord3(parseRecord(records.get(i++), PsseTransformer.ThirdRecord.class));
+    private static String readLineAndRemoveComment(BufferedReader reader) throws IOException {
+        String line = reader.readLine();
+        if (line == null) {
+            return null;
```

Comment:
From my point of view, if `readLine` returns null, that means we are at the end of the file, whereas we wanted to read more data. Maybe a premature end of file? In that case, should we throw an exception?

Reply:
Done.  I throw an exception.